lib/txnManager.js

```diff
@@ -0,0 +1,89 @@
+const assert = require('bsert');
+
+const TXUX = require('./uxtx.js');
+
+class TxnManager {
+  constructor(options) {
+    // hash -> parsed transaction
+    this._parsed = {};
+
+    if (options)
+      this.fromOptions(options);
+  }
+
+  static fromOptions(options) {
+    return new this().fromOptions(options);
+  }
+
+  fromOptions(options) {
+    if (options.custom) {
+      assert(typeof options.custom === 'function');
+      this.custom = options.custom;
+    }
+
+    if (options.constants) {
+      // TODO: validation
+      this.constants = options.constants;
+    }
+
+    return this;
+  }
+
+  /**
+   * Clear any cached values.
+   */
+  // TODO: _ prefix
+  refresh() {
+    this._parsed = {};
+  }
+
+  // bust - bool
+  // update api to allow for list of txns to bust
+  parse(transactions, wallet = null, bust = false) {
+    // use custom parsing function
```

Comment:
Need to make sure that transactions is always an array and that if there is a custom function it returns transactions as an array too.